### ISSUE TITLE
[import-order] order within `import type`

### ISSUE THREAD

**Issue by alvis** [import-js/eslint-plugin-import#2172]
Now, `import type` can be a separate group.
I also find the ability to alphabetize very useful.

However, I find it very annoying that they don't work well together.
The problem is that the order of `import type` statement doesn't follow the order we define for other module types.
Let's say we have the following options

```json
"import/order": [
  "warn",
  {
    "groups": [
      "builtin",
      "external",
      "type"
    ],
    "newlines-between": "always-and-inside-groups",
    "alphabetize": { "order": "asc", "caseInsensitive": true }
  }
],
```

e.g. I'd like to have the following order
```ts
// buildin modules before external modules

// for ordinary import it's ok, builtin first, then external
import { createServer } from 'http'

import { graphql } from 'gatsby';

// but for `import type` it warns that the gatsby import should be put in front of the http one
// the order is right here since http is a buildin and gatsby is an external
import type { RequestOptions } from 'http'

import type { PageProps } from 'gatsby';
```

**Comment by ljharb** [comment]
Seems like to do what you want, the schema would need some way of providing alternative config options for a specific group.

I suppose we could add `groupOverrides: { [groupName]: groupConfiguration }`, but that seems like a lot of complexity for such a specific use case.

**Comment by alvis** [comment]
or can we take the order from `groups` excluding `type` when we analyze the order within `import type` statements?

**Comment by ljharb** [comment]
I'm not sure what you mean. Can you elaborate?

**Comment by Exelenz9** [comment]
[This answer](https://github.com/import-js/eslint-plugin-import/issues/2088#issuecomment-848590569) helped me to solve similar problem

**Comment by kaysonwu** [comment]
I have the same issue. @ljharb the `type` group should also follow the basic order, e.g. : "builtin", "external", "internal", "parent", "Sibling ","index", "object"

**Comment by danranVm** [comment]
> Seems like to do what you want, the schema would need some way of providing alternative config options for a specific group.
> 
> I suppose we could add , but that seems like a lot of complexity for such a specific use case.`groupOverrides: { [groupName]: groupConfiguration }`

Hope to support this configuration, thank you.
